run/markdown-preview/editor/render.js

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const {GoogleAuth} = require('google-auth-library');
+const got = require('got');
+const auth = new GoogleAuth();
+
+let client;
+
+// renderRequest creates a new HTTP request with IAM ID Token credential.
+// This token is automatically handled by private Cloud Run (fully managed) and Cloud Functions.
+const renderRequest = async (markdown) => { 
+  // [START run_secure_request]
+  const serviceUrl = process.env.EDITOR_UPSTREAM_RENDER_URL;
```

Comment:
We still need a check that this exists. @grayside would this be best to check for on start up?

Reply:
Done - check implemented within the renderRequest function:
```
let serviceUrl;
...
const renderRequest = async (markdown) => { 
  if (process.env.EDITOR_UPSTREAM_RENDER_URL) serviceUrl = process.env.EDITOR_UPSTREAM_RENDER_URL;
```
My thought is that I wouldn't want my whole app to not run if one piece was missing. However, for the sake of this sample, perhaps I do want an error thrown if the RENDER_URL doesn't exist when the app is spun up, since the whole point of the app is to demonstrate service-to-service authentication. Feedback?